### ISSUE TITLE
lin estimator should weight before scaling

### ISSUE THREAD

**Issue by lukesonnet** [DeclareDesign/estimatr#148]

**Comment by lukesonnet** [comment]
I believe I have this implemented appropriately now. The following test passes:

```r
  lmwo <- lm_lin(mpg ~ am, ~ hp, weights = wt, data = mtcars)
  hp_wmean <- weighted.mean(mtcars$hp, mtcars$wt)
  wareg <- lm(mpg ~ hp, data = mtcars, subset = am == 1, weights = wt)
  wbreg <- lm(mpg ~ hp, data = mtcars, subset = am == 0, weights = wt)
  wate <-
    with(mtcars[mtcars$am == 1, ],
         weighted.mean(mpg, wt) + (hp_wmean - weighted.mean(hp, wt)) * wareg$coefficients[2]) -
    with(mtcars[mtcars$am == 0, ],
         weighted.mean(mpg, wt) + (hp_wmean - weighted.mean(hp, wt)) * wbreg$coefficients[2])

  expect_equivalent(
    wate,
    lmwo$coefficients["am"]
  )
```

However, the following is no longer true:
```r
  dat <- data.frame(Y = rnorm(4))
  dat$bl <- rep(1:25, each = 16)
  dat$z_blocked <- rep(c(0, 1), each = 2)
  dat$w <- runif(nrow(dat))
  expect_equivalent(
    tidy(lm_lin(Y ~ z_blocked, ~ factor(bl), weights = w, data = dat))[2, ],
    tidy(difference_in_means(Y ~ z_blocked, data = dat, blocks = bl, weights = w))
  )
```

I think this is because I'm not properly combining weighted block means. Will investigate right now.

**Comment by lukesonnet** [comment]
Closed by #169
